paimon-core/src/main/java/org/apache/paimon/iceberg/metadata/IcebergDataField.java

```diff
@@ -137,38 +148,23 @@ private static String toTypeString(DataType dataType) {
                 DecimalType decimalType = (DecimalType) dataType;
                 return String.format(
                         "decimal(%d, %d)", decimalType.getPrecision(), decimalType.getScale());
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                int timestampPrecision = ((TimestampType) dataType).getPrecision();
+                Preconditions.checkArgument(
```

Comment:
There are four place check this precision.
Should we extract this to reduce common code?